packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -60,22 +75,35 @@ export const getThorTradeQuote: GetThorTradeQuote = async ({ deps, input }) => {
     const sellAdapter = deps.adapterManager.get(chainId)
     const buyAdapter = deps.adapterManager.get(buyAssetChainId)
     if (!sellAdapter || !buyAdapter)
-      throw new SwapError(
-        `[getThorTradeQuote] - No chain adapter found for ${chainId} or ${buyAssetChainId}.`,
-        {
+      return Err(
+        makeSwapErrorMonad({
+          message: `[getThorTradeQuote] - No chain adapter found for ${chainId} or ${buyAssetChainId}.`,
           code: SwapErrorType.UNSUPPORTED_CHAIN,
           details: { sellAssetChainId: chainId, buyAssetChainId },
-        },
+        }),
       )
 
-    const rate = await getTradeRate({
+    const quoteRate = await getTradeRate({
       sellAsset,
       buyAssetId: buyAsset.assetId,
       sellAmountCryptoBaseUnit,
       receiveAddress,
       deps,
     })
 
+    const rate = await quoteRate.match<Promise<string> | string>({
+      ok: rate => rate,
+      // TODO: Handle TRADE_BELOW_MINIMUM specifically and return a result here as well
+      // Though realistically, TRADE_BELOW_MINIMUM is the only one we should really be seeing here,
+      // safety never hurts
+      err: _err =>
+        getTradeRateBelowMinimum({
+          sellAssetId: sellAsset.assetId,
+          buyAssetId: buyAsset.assetId,
+          deps,
+        }),
+    })
```

Comment:
The guts of this PR really @woodenfurniture - this is what we're eventually solving

Reply:
`quoteRate.match<Promise<string> | string>` looks really confusing but probably not something we can address

Reply:
Agree it's a bit ugly, I guess promisifying the ok side would make it consistent (at the cost of a useless promise job in the microtask queue) i.e:

```typescript
      ok: rate => Promise.resolve(rate),
```

Reply:
Actually consistent is probably better, same thing like we consistently `await` certain methods even if they're synchronous - 26f570b266f0585d84ff9cdb3b7ca2e564b7d779 ty ser!